jdaviz/configs/imviz/plugins/coords_info/coords_info.py

```diff
@@ -457,8 +481,11 @@ def _image_viewer_update(self, viewer, x, y):
                 unit = image.get_component(attribute).units
             elif isinstance(viewer, CubevizImageView):
                 arr = image.get_component(attribute).data
-                unit = image.get_component(attribute).units
-                value = self._get_cube_value(image, arr, x, y, viewer)
+                unit = self.current_unit
+                value = self._get_cube_value(
+                    image, arr, x, y, viewer
+                )._convert_surface_brightness_units(value, self.previous_unit, unit)
```

Comment:
I think this will fail once more than one conversion happens.  We want to convert from the native units of the data to the display units.  If the conversion is expensive from astropy, maybe we can instead cache the scale factor internally somehow (but that can be a follow-up if necessary).